docs/src/catalyst_applications/jump_simulation_performance.md

```diff
@@ -0,0 +1,163 @@
+
+Next, we provide the `save_positions = (false, false)` option to `JumpProblem`. This turns off the saving of the solution both before and after each jump. 
+```@example jump_simulation_performance_1
+jprob = JumpProblem(rn, dprob, Direct(); save_positions = (false, false))
+nothing # hide
+```
+However, if we were to simulate our model now, the solution would only actually be saved at its initial and final times. To remedy this, we provide the `saveat = 1.0` argument to the `solve` command, causing the solution to be saved at every `1.0`th time unit (at `t` = `0.0`, `1.0`, `2.0`, ... `999.0`, `1000.0`).
+```@example jump_simulation_performance_1
+sol = solve(jprob, SSAStepper(),saveat=1.0)
+nothing # hide
+```
+we can now plot the new solution:
+```@example jump_simulation_performance_1
+@btime plot(sol)
+```
+Here, we note that the time to plot the simulation was reduced (for this example, admittedly from an already low level). Furthermore, the plots look different since the trajectory is sampled at much sparser time points.
+
+## Types of jumps
+Each reaction in a chemical reaction network model corresponds to a possible jump of the jump simulation. These jumps can be divided into 3 categories:
+- `MassActionJump`s: These correspond to reactions which rates are constant throughout the simulation. They are typically generated when the rate contains parameters only. 
+- `ConstantRateJump`s: These correspond to reactions which rates remain constant between individual jumps, but may change in response to a jump occurring. They are typically generated when the rate contains variables and parameters only. 
+- `VariableRateJump`s: These correspond to reactions which rates may change at any time during the simulation. They are typically generated when the rate depends on the time variable ($t$).
+
+Here are some example reactions for the different types of jumps:
+```@example jump_simulation_performance_1
+# `MassActionJump`s
+@reaction 1.0, X --> Y
+@reaction k, 2X + Y --> X2Y
+@reaction k1*k2+k3, 0 --> X
+
+# `ConstantRateJump`s
+@reaction k*log(X), Y --> X
+@reaction mm(X,v,K), 0 --> X
+
+# `VariableRateJump`s
+@reaction k*(1+sin(t)), 0 --> X
+@reaction X/t, X + Y --> XY
+nothing # hide
+```
+Throughout a simulation, `VariableRateJump`s' rates require updating more frequently than `ConstantRateJump`s' rates, which in turn requires updating more frequently than `MassActionJump`s' rates. Since the performance of jump simulations is heavily affected by how frequently jump rates are computed, keeping the number of `ConstantRateJump`s and `VariableRateJump`s as small as possible is advantageous. Primarily, there exist two common cases where models are written in a way so that sub-optimal jump types are generated, both of which we describe below.
```

Comment:
```Throughout a simulation, `VariableRateJump`s' rates require updating more frequently than `ConstantRateJump`s' rates, which in turn requires updating more frequently than `MassActionJump`s' rates. ```
This isn't really the issue (or even true). I would more say that evaluating propensities and state changes due to `MassActionJump`s are more computationally efficient due to their constrained form than updating them for `ConstantRateJump`s (which are more general in the propensity functions and state changes they support but lose computational efficiency due to this flexibility). `VariableRateJump`s require more computationally expensive simulation algorithms than standard SSAs, which require root finding to calculate next event times (except when using rejection-based approaches via rate bounds, but Catalyst doesn't support that yet so it probably isn't worth mentioning).